bitnami/harbor/templates/clair/clair-dpl.yaml

```diff
@@ -3,105 +3,174 @@ apiVersion: {{ template "deployment.apiVersion" . }}
+            successThreshold: {{ .Values.clair.server.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.clair.server.livenessProbe.failureThreshold }}
+          {{- else if .Values.clair.server.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.clair.server.customLivenessProbe "context" $) | nindent 12 }}
           {{- end }}
-          {{- if .Values.clair.readinessProbe.enabled }}
+          {{- if .Values.clair.server.readinessProbe.enabled }}
           readinessProbe:
             httpGet:
               path: /health
               port: 6061
-            initialDelaySeconds: {{ .Values.clair.readinessProbe.initialDelaySeconds }}
-            periodSeconds: {{ .Values.clair.readinessProbe.periodSeconds }}
-            timeoutSeconds: {{ .Values.clair.readinessProbe.timeoutSeconds }}
-            successThreshold: {{ .Values.clair.readinessProbe.successThreshold }}
-            failureThreshold: {{ .Values.clair.readinessProbe.failureThreshold }}
+            initialDelaySeconds: {{ .Values.clair.server.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.clair.server.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.clair.server.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.clair.server.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.clair.server.readinessProbe.failureThreshold }}
+          {{- else if .Values.clair.server.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.clair.server.customReadinessProbe "context" $) | nindent 12 }}
           {{- end }}
           env:
             - name: BITNAMI_DEBUG
               value: {{ ternary "true" "false" .Values.clairImage.debug | quote }}
+            {{- if has "clair" .Values.proxy.components }}
             {{- if .Values.clair.httpProxy }}
             - name: HTTP_PROXY
-              value: {{ .Values.clair.httpProxy }}
+              value: {{ .Values.clair.httpProxy | quote }}
             {{- end }}
             {{- if .Values.clair.httpsProxy }}
             - name: HTTPS_PROXY
-              value: {{ .Values.clair.httpsProxy }}
+              value: {{ .Values.clair.httpsProxy | quote }}
             {{- end }}
             - name: NO_PROXY
-              value: "{{ include "harbor.registry" . }},{{ include "harbor.core" . }}"
-          {{- if .Values.clair.resources }}
-          resources: {{- toYaml .Values.clair.resources | nindent 12 }}
+              value: {{ include "harbor.noProxy" . | quote }}
+            {{- end }}
+            {{- if .Values.clair.server.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.clair.server.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- if .Values.clair.server.extraEnvVarsCM }}
```

Comment:
Shouldn't we include checksum to those configmapref and secretref?

Reply:
That's not something we can calculate the checksum because we do not render it. If the user generates it using extraDeploy or an umbrella chart, should add the checksum using podAnnotations.